core/server/services/settings/index.js

```diff
@@ -4,16 +4,78 @@
  */
 
 var SettingsModel = require('../../models/settings').Settings,
-    SettingsCache = require('./cache');
+    SettingsCache = require('./cache'),
+    SettingsLoader = require('./loader'),
+    _ = require('lodash'),
+    common = require('../../lib/common');
+    // SettingsFiles = require('./ensure-settings');
 
 module.exports = {
     init: function init() {
+        // TODO: the ensure settings fn up before the settings cache
+        // return SettingsFiles(this.knownSettings()).then(() => {
+
         // Update the defaults
         return SettingsModel.populateDefaults()
             .then(function (settingsCollection) {
                 // Initialise the cache with the result
                 // This will bind to events for further updates
                 SettingsCache.init(settingsCollection);
             });
+        // });
+    },
+
+    // TODO: is this the right place for it? We need a global place where
```

Comment:
Yeah i guess that's fine. Not sure it has to be exposed though - probably an internal information of this service.